_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,428 @@
+
+To check the partitions details of a certain topic, we can use the `kafka-topics.sh` tool. We can run the following command from inside interactive pod after starting a shell process using `kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash`
+:
+
+```sh
+bin/kafka-topics.sh --describe --topic my-topic-two --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092
+```
+which will give us the following output:
+
+```shell
+Topic: my-topic-two     TopicId: _aUFY9oMSjqBjvP9Ed3JDg PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic-two     Partition: 0    Leader: 0       Replicas: 0,3,1 Isr: 0,3,1
+        Topic: my-topic-two     Partition: 1    Leader: 1       Replicas: 1,0,4 Isr: 1,0,4
+        Topic: my-topic-two     Partition: 2    Leader: 4       Replicas: 4,1,2 Isr: 4,1,2
+        Topic: my-topic-two     Partition: 3    Leader: 2       Replicas: 2,4,3 Isr: 2,4,3
+        Topic: my-topic-two     Partition: 4    Leader: 3       Replicas: 3,2,0 Isr: 3,2,0
+        Topic: my-topic-two     Partition: 5    Leader: 0       Replicas: 0,1,4 Isr: 0,1,4
+        Topic: my-topic-two     Partition: 6    Leader: 1       Replicas: 1,4,2 Isr: 1,4,2
+        Topic: my-topic-two     Partition: 7    Leader: 4       Replicas: 4,2,3 Isr: 4,2,3
+        Topic: my-topic-two     Partition: 8    Leader: 2       Replicas: 2,3,0 Isr: 2,3,0
+        Topic: my-topic-two     Partition: 9    Leader: 3       Replicas: 3,0,1 Isr: 3,0,1
+```
+Note: Your topic details can vary and might not be same as the topic details present here. 
+
+In the same way you can get the details for the other topic `my-topic` also.
+
+```sh
+Topic: my-topic TopicId: WyFKVZzLS8i54IGgm1ifrQ PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic Partition: 0    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
+        Topic: my-topic Partition: 1    Leader: 2       Replicas: 2,1,0 Isr: 0,2,1
+        Topic: my-topic Partition: 2    Leader: 1       Replicas: 1,0,2 Isr: 0,2,1
+        Topic: my-topic Partition: 3    Leader: 0       Replicas: 0,1,2 Isr: 0,2,1
+        Topic: my-topic Partition: 4    Leader: 2       Replicas: 2,0,1 Isr: 0,2,1
+        Topic: my-topic Partition: 5    Leader: 1       Replicas: 1,2,0 Isr: 0,2,1
+        Topic: my-topic Partition: 6    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
+        Topic: my-topic Partition: 7    Leader: 2       Replicas: 2,1,0 Isr: 0,2,1
+        Topic: my-topic Partition: 8    Leader: 1       Replicas: 1,0,2 Isr: 0,2,1
+        Topic: my-topic Partition: 9    Leader: 0       Replicas: 0,1,2 Isr: 0,2,1
+```
+
+From the above outputs, we got to know that my-topic-two has some replicas on broker 3 and 4 thus we need to reassign this topic onto other brokers. 
```

Comment:
`we got to know that my-topic-two` -> `we see that my-topic-two` and put backticks around `my-topic-two`